lib/hammer_cli_foreman/compute_profile.rb

```diff
@@ -0,0 +1,423 @@
+              if defined?(provider.interface_create_help)
+                provider.interface_create_help(h)
+              end
+            end
+          end
+        end
+      end
+
+      def validate_options
+        super
+        validator.any(:option_compute_profile_id,:option_compute_profile_name).required
+        validator.any(:option_compute_resource_id,:option_compute_resource_name).required
+        validator.all(:option_set_values).required
+      end
+
+      def request_params
+        params = HammerCLIForeman::ComputeProfile.get_params(options)
+        interface_attr =  params['compute_attribute']['vm_attrs']['interfaces_attributes'] || {}
+        params['id'] = params['compute_attribute']['id']
+        params['compute_attribute']['vm_attrs']['interfaces_attributes'] = interface_attr.merge({Time.now.to_i.to_s => option_set_values})
+        params
+      end
+
+      success_message _('Interface was created')
+      failure_message _('Could not create interface')
+
+      build_options :without => :id
+    end
+
+    class UpdateInterface < HammerCLIForeman::UpdateCommand
+      command_name _('update-interface')
+      resource :compute_attributes
+
+      desc _('Update compute profile interface')
+
+      option '--set-values', 'SET VALUES', _('Interface parameters, should be comma separated list of values'),
+             :required => true,
+             :format => HammerCLI::Options::Normalizers::KeyValueList.new
+      option '--interface-id', 'INTERFACE_ID', _('Interface id'),
+             :required => true,
+             :format => HammerCLI::Options::Normalizers::Number.new
+      extend_help do |h|
+        h.section _('Provider Interface specific options') do |h|
+          ::HammerCLIForeman::ComputeProfile.compute_resources.each do |name, provider|
+            h.section name do |h|
+              if defined?(provider.interface_create_help)
+                provider.interface_create_help(h)
+              end
+            end
+          end
+        end
+      end
+
+      def validate_options
+        super
+        validator.any(:option_compute_profile_id, :option_compute_profile_name).required
+        validator.any(:option_compute_resource_id, :option_compute_resource_name).required
+        validator.all(:option_interface_id, :option_set_values).required
+      end
+
+      def request_params
+        params = HammerCLIForeman::ComputeProfile.get_params(options)
+        interface_attr =  params['compute_attribute']['vm_attrs']['interfaces_attributes'] || {}
+        params['id'] = params['compute_attribute']['id']
+        params['compute_attribute']['vm_attrs']['interfaces_attributes'] = interface_attr.merge({option_interface_id => option_set_values})
+        params
+      end
+      success_message _('Interface was updated')
+      failure_message _('Could not update interface')
+      build_options :without => :id
+    end
+
+    class RemoveInterface < HammerCLIForeman::UpdateCommand
```

Comment:
Maybe it should inherit `DeleteCommand` instead of `UpdateCommand`?

Reply:
I think the UpdateCommand is needed in this case because the remove is done by updating the resource.